src/Poseidon/Janno.hs

```diff
@@ -97,6 +98,30 @@ instance Show JannoSex where
     show (JannoSex Male)    = "M"
     show (JannoSex Unknown) = "U"
 
+-- | A datatype for BC-AD ages
+newtype BCADAge =
+        BCADAge Int
+    deriving (Eq, Ord, Generic)
+
+instance ToJSON BCADAge where
+    toEncoding = genericToEncoding defaultOptions
+
+instance FromJSON BCADAge
+
+instance Csv.FromField BCADAge where
+    parseField x = do
+        val <- Csv.parseField x
+        if val >= 2022 -- the current year
```

Comment:
Haha, now we have a once-per-year maintenance task to do here :-) Better get the current year programmatically, right? I'll quickly look into it.

Reply:
Ah - right - that was stupid on my end :facepalm: